### ISSUE TITLE
Issue with React Context API

### ISSUE THREAD

**Issue by stereobooster** [airbnb/enzyme#1693]
**Describe the bug**

This is follow up for #1513 (#1509). I copy-pasted adapter from master branch. I see an error

```
  console.error node_modules/fbjs/lib/warning.js:33
    Warning: Detected multiple renderers concurrently rendering the same context provider. This is currently unsupported.
```

**To Reproduce**
1. `git clone https://github.com/stereobooster/react-context-issue`
2. `cd react-context-issue`
3. `yarn`
4. `yarn test`

**Expected behavior**
Ability to run multiple renderers concurrently

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: MacOS X, node 8.6
 - Browser N/A
 - Version N/A

**Smartphone (please complete the following information):**
 N/A

**Additional context**

Related: https://github.com/facebook/react/issues/12589, https://github.com/facebook/jest/issues/6183

**Comment by stereobooster** [comment]
As far as I understand to fix this issue we need to "undo" fiber batch rendering and render inside test in one piece, without batches.

**Comment by ljharb** [comment]
This sounds like you're having an issue with unreleased code?

**Comment by stereobooster** [comment]
Yes this is the issue with unreleased code. It is meant to be added to the list in https://github.com/airbnb/enzyme/issues/1553

**Comment by stereobooster** [comment]
Huh. I figured out what the issue here. It happens when you use `react-test-renderer` and Enzyme's `mount` in the same test file. Not sure if it is needed to be fixed...

**Comment by stereobooster** [comment]
To be clear if I use only `react-test-renderer` it is ok. If I use only `Enzyme` it is ok. Issue happens when I try to use `react-test-renderer` (for snapshot testing) and `enzyme` (for actual testing, like simulating clicking buttons etc.)

I found solution: use `enzyme-to-json` instead of `react-test-renderer` for snapshot testing.

Example of how I fixed it is [here](https://github.com/stereobooster/react-context-issue/tree/enzyme-to-json).
